components/geolocation/geolocation.service.ts

```diff
@@ -0,0 +1,338 @@
+    /**
+     * @ngdoc property
+     * @name HsGeolocationService#following
+     * @public
+     * @type {boolean} false
+     * @description Represents geolocalization tracking option (on/off).
+     * Used to deremine state of tracking in directive's html
+     */
+    this.following = false;
+    this.gn = null;
+    /**
+     * @ngdoc method
+     * @name HsGeolocationService#stopCentering
+     * @public
+     * @description Turns off position centering while 'following'.
+     */
+    this.stopCentering = this.HsUtilsService.debounce(
+      () => {
+        this.centering = false;
+      },
+      150,
+      false,
+      this
+    );
+
+    this.accuracyFeature = new Feature({
+      known: false,
+      geometry: new CircleGeom([0, 0], 1),
+    });
+    this.positionFeature = new Feature({
+      known: false,
+      geometry: new Point([0, 0]),
+    });
+    this.style = new Style({
+      image: new Circle({
+        fill: new Fill({
+          color: [242, 121, 0, 0.7],
+        }),
+        stroke: new Stroke({
+          color: [0xbb, 0x33, 0x33, 0.7],
+        }),
+        radius: 5,
+      }),
+      fill: new Fill({
+        color: [0xbb, 0xbb, 0xbb, 0.2],
+      }),
+      stroke: new Stroke({
+        color: [0x66, 0x66, 0x00, 0.8],
+      }),
+    });
+    this.HsMapService.loaded().then((map) => this.init(map));
+  }
+  /**
+   * @ngdoc method
+   * @name HsGeolocationService#stopTracking
+   * @public
+   * @description Reset all geolocalization parameters concerning position tracking
+   */
+  stopTracking(): void {
+    this.following = false;
+    this.HsMapService.map
+      .getControls()
+      .getArray()[4]
+      .element.classList.add('hidden');
+    this.HsMapService.map.on('pointermove', this.stopCentering);
+    this.geolocation.setTracking(false);
+    if (this.gn !== null) {
+      this.gn.stop();
+    }
+    this.HsMapService.map.getView().setRotation(0);
+  }
+
+  /**
+   * @ngdoc method
+   * @name HsGeolocationService#stopTracking
+   * @public
+   * @description Toggles tracking/following
+   * Takes care of the distinction between click and double-click on mobile
+   */
+  toggleTracking(): any {
+    if (this.clicked) {
+      this.cancelClick = true;
+      if (this.HsLayoutService.sidebarBottom()) {
+        this.HsLayoutService.contentWrapper
+          .querySelector('.hs-locationButton')
+          .dispatchEvent(new Event('dblclick'));
+      }
+      return;
+    }
+    this.clicked = true;
+    setTimeout(() => {
+      if (this.cancelClick) {
+        this.cancelClick = false;
+        this.clicked = false;
+        return;
+      }
+      if (this.isCentered()) {
+        if (!this.following) {
+          //position
+          this.geolocation.on('change:position', this.setNewPosition());
+          this.geolocation.setTracking(true);
+          this.following = true;
+          //rotation
+          this.setRotation();
+          this.geolocation.on('change:heading', this.newRotation());
```

Comment:
Is it working more then once? Should it not be  () => this.newRotation() ?

Reply:
When I do as You suggest the map starts to flicker all the time

Reply:
Fixed map flickering with HsUtilsService debounce method